libcaf_core/caf/json_reader.test.cpp

```diff
@@ -388,6 +424,22 @@ fixture::fixture() {
   // Test cases for proper handling of whitespace.
   add_test_case("[1, \r\n2,\f\t\v3]", ls<int32_t>(1, 2, 3));
   add_test_case("\r\n{\r\n\"a\":\r\n1, \"b\"\r\n:\r\n2}\r\n", my_request(1, 2));
+  // Test cases for proper handling of code point escape sequences.
+  // Single byte utf-8 sequences.
+  add_test_case(
+    R"_("\u0048\u0065\u006c\u006c\u006f\u002c\u0020\u0022\u0057\u006f\u0072\u006c\u0064\u0022\u0021")_",
+    std::string{R"_(Hello, "World"!)_"});
+  // TODO why do we need two backslashes when the first one is already escaped?
```

Comment:
What's left to do here?

Reply:
This one got me thinking. Shouldn't two backslashes, when escaped, `\u005c\u005c` resolve to two backslashes `std::string{R"_(\\)_`? Any opinions?

Reply:
Why should it not? `\u005c\u005c` is basically equivalent to `\\\\`, no? In both cases, I would expect the unescaped string to be `\\`.